src/stirling/obj_tools/elf_reader.cc

```diff
@@ -614,6 +614,15 @@ StatusOr<utils::u8string> ElfReader::SymbolByteCode(std::string_view section,
   if (!ifs.seekg(offset)) {
     return error::Internal("Failed to seek position=$0 in binary=$1", offset, binary_path_);
   }
+  // To protect against our ELF parsing logic locating bogus memory, set a bound on
+  // how large of a string we will allocate (100 MB). See
+  // https://github.com/pixie-io/pixie/issues/1111 for more details.
+  if (symbol.size > 100 * 1024 * 1024) {
```

Comment:
Do you think it makes sense to try and address the bug that causes the symbol size to be so large, rather than just protecting against it?